Azure/sonic-buildimage
# [Monit] Unmonitor the processes in containers which are disabled.

**- Why I did it**
We want to let Monit to unmonitor the processes in containers which are disabled in `FEATURE` table such that
Monit will not generate false alerting messages into the syslog.

**- How I did it**
Monit will periodically run a script which accepts three parameters: `<container_name>`, `<process_name>` and
`<process_cmdline>`. This script will first check whether the container is disabled in the `FEATURE` table or not.
If it is disabled, Monit will skip monitoring the processes. Otherwise, this script will leverage psutil library to inspect
the process tree in host to look for the processes. If the process is not found, then an alerting message will be written
into syslog.

**- How to verify it**
We can change the `state` field of a container in `FEATURE` table from `enabled` to `disabled` and then kill a critical 
process in it to see whether Monit can generate the alerting message in syslog or not. The message format in syslog is:
`<process_name> is not running.`

**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
